### ISSUE TITLE
register doesn't work

### ISSUE THREAD

**Issue by Mortuie** [chingu-voyage3/bears-21#114]
See live page

**Comment by Mortuie** [comment]
```==============request user:undefined
[0] (node:9574) DeprecationWarning: collection.update is deprecated. Use updateOne, updateMany, or bulkWrite instead.
[0] GET /api/v1/user 401 12 - 29.919 ms
[0] 
[0] ==============request user:undefined
[0] POST /api/v1/register 200 68 - 2078.324 ms
[0] 
[0] ==============request user:{ email: 'j@bob.com',
[0]   _id: 5ba29e4e428c562566e9c32f,
[0]   name: 'leon',
[0]   __v: 0 }
[0] POST /api/v1/register 400 104 - 26.423 ms

**Comment by Mortuie** [comment]
This seems to be the stack trace on the backend

**Comment by nikrb** [comment]
is that live or local? on local I get:
```
[0] ==============request user:undefined
[0] POST /api/v1/register 200 57 - 2238.076 ms
```

**Comment by nikrb** [comment]
oh it looks like you tried to register same user twice? ... or maybe tried registering whilst logged in - I haven't tried that.

**Comment by Mortuie** [comment]
Local

**Comment by Mortuie** [comment]
I mean if we're registering with the same user twice we should also give a useful error message :P

**Comment by vivekimsit** [comment]
Yes, [here](https://github.com/chingu-voyage3/bears-21/blob/master/web/routes/users/register.js#L37) should be a check before saving the user.

**Comment by Mortuie** [comment]
I mean on the frontend as well not just the backend..

**Comment by vivekimsit** [comment]
Well to do that you need to make a XHR call to check if username is already present.

**Comment by Mortuie** [comment]
Yusss I know :P, basically both are missing xD

**Comment by nikrb** [comment]
how about we just display the error message returned from register (rather than making *another* xhr call)? Hmmm, is [validateRegister](https://github.com/chingu-voyage3/bears-21/blob/f55c4dd5282f4f8670bd9a65b4221ec6d91379d8/web/routes/users/register.js#L6) being called?

**Comment by Mortuie** [comment]
On live we now have ```POST http://hissues.herokuapp.com/api/v1/register 400 (Bad Request)``` when trying to register and ```POST http://hissues.herokuapp.com/api/v1/register 500 (Internal Server Error)```

**Comment by nikrb** [comment]
well let's get develop fixed up and then we can merge to master and update live.

**Comment by nikrb** [comment]
register still working for me
